src/test/java/teammates/test/cases/testdriver/HtmlHelperTest.java

```diff
@@ -135,4 +135,20 @@ public void testConvertToStandardHtmlPart() throws Exception {
         
         //other cases are tested in testComparison
     }
+    
+    @Test
+    public void testReplacement() throws IOException {
```

Comment:
My only concern is that we are testing right at the end, after processing both the expected and actual html. If we can do one more check (for instance after the ```injectTestProperties```), it strengthens the test we are conducting. This check can be another html compare (just create the html file after injectTestProperties). Since we are conducting the test **without** any processing of the "expected" html, the test is more robust.

I understand that it is pointless testing ```injectContextDependentValuesForTest``` as that function was made for this purpose!

Bottom Line: We are testing whether the output of the processing on both the sides yields the same result. BUT we are not checking whether this result is the CORRECT result. They can be the same, but incorrect.